permissions/src/main/java/org/openmbee/sdvc/permissions/DefaultPermissionService.java

```diff
@@ -82,102 +86,122 @@ public void initBranchPerms(String projectId, String branchId, boolean inherit,
         Project project = getProject(projectId);
         PermissionsDelegate permissionsDelegate = getPermissionsDelegate(project);
-        permissionsDelegate.updateUserPermissions(req);
+        PermissionUpdatesResponseBuilder responseBuilder = new PermissionUpdatesResponseBuilder();
+        responseBuilder.getUsers().insert(permissionsDelegate.updateUserPermissions(req));
 
         for (Branch b : project.getBranches()) {
-            recalculateInheritedPerms(b);
+            responseBuilder.insert(recalculateInheritedPerms(b));
         }
+
+        return responseBuilder.getPermissionUpdatesReponse();
     }
 
     @Override
     @Transactional
-    public void updateProjectGroupPerms(PermissionUpdateRequest req, String projectId) {
+    public PermissionUpdatesResponse updateProjectGroupPerms(PermissionUpdateRequest req, String projectId) {
         Project project = getProject(projectId);
         PermissionsDelegate permissionsDelegate = getPermissionsDelegate(project);
-        permissionsDelegate.updateGroupPermissions(req);
+        PermissionUpdatesResponseBuilder responseBuilder = new PermissionUpdatesResponseBuilder();
+        responseBuilder.getGroups().insert(permissionsDelegate.updateGroupPermissions(req));
 
         for (Branch b : project.getBranches()) {
-            recalculateInheritedPerms(b);
+            responseBuilder.insert(recalculateInheritedPerms(b));
         }
+
+        return responseBuilder.getPermissionUpdatesReponse();
     }
 
     @Override
     @Transactional
-    public void updateBranchUserPerms(PermissionUpdateRequest req, String projectId, String branchId) {
+    public PermissionUpdateResponse updateBranchUserPerms(PermissionUpdateRequest req, String projectId, String branchId) {
         Branch branch = getBranch(projectId, branchId, BRANCH_NOTFOUND_BEHAVIOR.THROW);
         PermissionsDelegate permissionsDelegate = getPermissionsDelegate(branch);
-        permissionsDelegate.updateUserPermissions(req);
+        return permissionsDelegate.updateUserPermissions(req);
     }
 
     @Override
     @Transactional
-    public void updateBranchGroupPerms(PermissionUpdateRequest req, String projectId, String branchId) {
+    public PermissionUpdateResponse updateBranchGroupPerms(PermissionUpdateRequest req, String projectId, String branchId) {
         Branch branch = getBranch(projectId, branchId, BRANCH_NOTFOUND_BEHAVIOR.THROW);
         PermissionsDelegate permissionsDelegate = getPermissionsDelegate(branch);
-        permissionsDelegate.updateGroupPermissions(req);
+        return permissionsDelegate.updateGroupPermissions(req);
     }
 
     @Override
     @Transactional
-    public void setProjectInherit(boolean isInherit, String projectId) {
+    public PermissionUpdatesResponse setProjectInherit(boolean isInherit, String projectId) {
+        PermissionUpdatesResponseBuilder responseBuilder = new PermissionUpdatesResponseBuilder();
+        responseBuilder.setInherit(true);
```

Comment:
there's several places where setInherit is hardcoded to true, think it should just be the parameter isInherit? Same for setOrg or setProject public (the method name is a bit misleading since it's actually setting it based on an argument

Reply:
actually if it's just to keep track of whether it's been set/processed then nevermind